### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by matiasgarcia91**
try to improve your use of branches 😄

### CODE REVIEW

File: client/src/App.css [hunk 1]
```diff
@@ -1,38 +1,193 @@
...
+    margin-top: 1rem;
+    min-height: 95vh;
   }
-  to {
-    transform: rotate(360deg);
+  .roomcards {
+    display: inline-block;
+    width: 80%;
   }
 }
```
Comment (matiasgarcia91) [inline client/src/App.css]: split this css file into its concerns, one for the form, etc

File: client/src/App.js [hunk 2]
```diff
@@ -1,8 +1,34 @@
...
+import "bootstrap/dist/css/bootstrap.min.css";
+import { Provider } from "react-redux";
+import { Switch, Route } from "react-router";
+import HomePageContainer from "./components/HomePageContainer";
+import LoginContainer from "./components/user/LoginContainer";
+import SignupContainer from "./components/user/SignupContainer";
+import ProductContainer from "./components/Product/ProductContainer";
+import DocumentContainer from "./components/document/DocumentContainer";
+import Heading from "./components/Heading";
+import Footer from "./components/Footer";
```
Comment (matiasgarcia91) [inline client/src/App.js]: imports have an order to look nice:
1. Libraries and modules
2. My own files (components, actions or in the backend models/routers)
3. css files

File: client/src/components/Product/ProductContainer.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+export default function ProductContainer() {
+  const params = useParams(); // used to get params from the App.js where Route was defined
+  // console.log("the params", params);
+  const dispatch = useDispatch();
+
+  const state = useSelector(reduxState => {
+    return {
+      productState: reduxState.product
+    };
+  });
```
Comment (matiasgarcia91) [inline client/src/components/Product/ProductContainer.js]:
```
 const state = useSelector(reduxState => ({ productState: reduxState.product })
```

File: client/src/store/product/action.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import axios from "axios";
```
Comment (matiasgarcia91) [inline client/src/store/product/action.js]: If you are using axios you can create an instance of it in a separate file with configurations and then export that instance and use that in your action files, like:
```
// src/axios.js
import axios from "axios";

const instance = axios.create({
  baseURL: "http://localhost:4000"
});
export default instance;
```

File: client/src/store/product/reducer.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        prod => prod.id !== action.payload.id
+      );
+      return { ...state, products: [...removeOldEdit, action.payload] };
+    }
+    default: {
+      return { ...state };
+    }
+  }
+};
+
```
Comment (matiasgarcia91) [inline client/src/store/product/reducer.js]: nice clean reducer 🥇

File: server/product/router.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
+const { Router } = require("express");
+const Sequelize = require("sequelize");
+const Product = require("./model");
+const auth = require("../auth/middleWare");
+const cron = require("node-cron");
+const moment = require("moment");
+const MailingList = require("../mailing/model");
+const sendMail = require("../sendMail");
```
Comment (matiasgarcia91) [inline server/product/router.js]: same thing about the import order

File: server/product/router.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+  });
+  const shootMail = MailingListArr.map(mailItem => {
+    const idtosendon = mailItem.dataValues.email_id;
+    sendMail(
+      idtosendon,
+      "Attention! your appliance warranty is expiring",
+      `Your product ID: ${mailItem.dataValues.product_id} is expiring in ${mailItem.dataValues.remaining_warranty} days`
+    );
+  });
+});
```
Comment (matiasgarcia91) [inline server/product/router.js]: maybe move the cron setup to it's own file? (nice job setting it up btw)

File: server/product/router.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+  } catch (error) {
+    next(console.error);
+  }
+});
+
+const findRemainingDays = product => {
+  const item = product.dataValues; //the item is a databse type object
+  const een = moment(item.warranty_end_date);
+  const twee = new Date();
+  const duration = moment.duration(een.diff(twee));
```
Comment (matiasgarcia91) [inline server/product/router.js]: useful eh? hahaha

File: server/room/router.js [hunk 9]
```diff
@@ -1,6 +1,59 @@
...
+
+    const deviceNameArr = productIdArr.map(async item => {
+      const prodItem = await Product.findByPk(item.productID);
+      item.deviceName = prodItem.device_name;
+      // console.log("--the newItem NOW -----------", item);
+      return item;
+    });
+
+    Promise.all(deviceNameArr).then(async () => {
+      // console.log("deviceNameArr--------------------", deviceNameArr);
```
Comment (matiasgarcia91) [inline server/room/router.js]: clean up consoles

File: server/testinglib.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+var moment = require("moment");
+var een = moment("2021-03-26 16:03:46");
+var twee = new Date();
+var duration = moment.duration(een.diff(twee));
```
Comment (matiasgarcia91) [inline server/testinglib.js]: a tip for this which is awesome. regularly these good libraries have the library itself loaded in their documentation webpage so if you open the browser console on momentjs.com you have `moment` available to play around with and test on the console.
Lodash the same

File: client/src/App.js (hunk 2 above)
Comment (learnerdilip) [reply client/src/App.js]: thats a good suggestion!

File: client/src/components/HomePageContainer.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+const HomePageContainer = () => {
+  return (
+    <div className="homepage">
+      <h2>Manage your Warranty and Invoices</h2>
+      <RoomContainer />
+    </div>
+  );
+};
+
+export default HomePageContainer;
```
[the remark this answers is not in the document]
Comment (learnerdilip) [reply client/src/components/HomePageContainer.js]: i agree

File: client/src/components/Product/ProductContainer.js [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+      setEditData(editableItem);
+    }
+  };
+
+  return (
+    <div className="productcontainer">
+      <h2>YOUR PRODUCTS LIST</h2>
+      <div>
+        {state.productState.products.length > 0 &&
+          state.productState.products.map(product => {
```
[the remark this answers is not in the document]
Comment (learnerdilip) [reply client/src/components/Product/ProductContainer.js]: yeah you are right, it actually looks crazy now ,  but it started small and it kept increasing :) (maybe i should move it to a different file once it gets big enough, i would like to discuss the best practice for this tho)

File: client/src/components/document/DocumentContainer.js [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+        <h5>THE NAME OF DOCUMENT: {document_name}</h5>
+        <h5>NAME ON INVOICE: {name_on_invoice} </h5>
+        <h5>DEVICE NAME: {device_name} </h5>
+        <h5>PURCHASE DATE: {moment(purchase_date).format("MMMM Do YYYY")} </h5>
+        <h5>
+          WARRANTY START DATE:{" "}
+          {moment(warranty_start_date).format("MMMM Do YYYY")}{" "}
+        </h5>
+        <h5>
+          WARRANTY END DATE: {moment(warranty_end_date).format("MMMM Do YYYY")}{" "}
```
[the remark this answers is not in the document]
Comment (learnerdilip) [reply client/src/components/document/DocumentContainer.js]: yes this is something that i felt during working on this as well, thank you!

File: client/src/store/product/action.js (hunk 4 above)
Comment (learnerdilip) [reply client/src/store/product/action.js]: i am not sure how this works, We can discuss this tomorrow maybe?

File: client/src/store/product/reducer.js (hunk 5 above)
Comment (learnerdilip) [reply client/src/store/product/reducer.js]: :)

File: server/product/router.js (hunk 7 above)
Comment (learnerdilip) [reply server/product/router.js]: yeah moving it makes sense to me too!

File: server/product/router.js (hunk 8 above)
Comment (learnerdilip) [reply server/product/router.js]: running out of variable names :D

File: server/testinglib.js (hunk 10 above)
Comment (learnerdilip) [reply server/testinglib.js]: thats super useful, never knew! thank you!

File: client/src/store/product/action.js (hunk 4 above)
Comment (matiasgarcia91) [reply client/src/store/product/action.js]: you just create a file like this and call it `src/axios.js` here you import the axios library and you can create an instance of it with some configs already set. Then you export that instance and from your action files you `import axios from '../axios.js'` and use straight away
```
`import axios from '../axios.js'`
axios.post('/products', { ... }).then(... etc)
```

**Pull request merged**